slimt/Frontend.hh

```diff
@@ -44,4 +46,28 @@ class Translator {
   size_t model_id_ = 0;
 };
 
+class Async {
+ public:
+  Async(const Config &config, View model, View shortlist, View vocabulary);
+  Response translate(std::string &source, const Options &options);
+  ~Async();
+
+ private:
+  Config config_;
+  Vocabulary vocabulary_;
+  TextProcessor processor_;
+  Model model_;
+  ShortlistGenerator shortlist_generator_;
+  std::optional<TranslationCache> cache_;
+  rd::Threadsafe<rd::Batcher> batcher_;
+  Histories forward(Batch &batch);
+  Histories decode(Tensor &encoder_out, Tensor &mask, const Words &source,
+                   const std::vector<size_t> &lengths);
```

Comment:
These two if I understand correctly are as-is duplicated with `Translator`. Can we achieve deduplication? Move the functions into an [anonymous namespace](https://stackoverflow.com/questions/357404/why-are-unnamed-namespaces-used-and-what-are-their-benefits) in `Frontend.cc` as [free functions](https://stackoverflow.com/questions/4861914/what-is-the-meaning-of-the-term-free-function-in-c)? Following this, both `Async` and `Translator` relays the call to the new free functions.